# Machine Learning ~ un proyecto ilustrativo.

Usaremos un Dataset que puede descargarse desde el siguiente link.
https://archive.ics.uci.edu/ml/datasets/seeds#

In [28]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

In [50]:
dataset = pd.read_csv('../850-MX-J/seeds_dataset.csv')   # Leemos nuestro archivo.
print('Shape: ', dataset.shape) # Esta linea nos dice cuantas filas y columnas tiene el Dataset.
dataset.sample(5)                                    # visualizamos 5 lineas aleatorias.

Shape:  (210, 8)


area      perimeter   compactness   length   width   coefficient   \
188     11.23       12.82        0.8594    5.089   2.821         7.524   
198     12.62       13.67        0.8481    5.410   2.911         3.306   
29      13.45       14.02        0.8604    5.516   3.065         3.531   
193     10.82       12.83        0.8256    5.180   2.630         4.853   
59      12.11       13.47        0.8392    5.159   3.032         1.502   

     length_groove   type_wheat  
188           4.957           3  
198           5.231           3  
29            5.097           1  
193           5.089           3  
59            4.519           1

In [51]:
dataset.describe()

area      perimeter   compactness      length       width   \
count  210.000000  210.000000    210.000000  210.000000  210.000000   
mean    14.847524   14.559286      0.870999    5.628533    3.258605   
std      2.909699    1.305959      0.023629    0.443063    0.377714   
min     10.590000   12.410000      0.808100    4.899000    2.630000   
25%     12.270000   13.450000      0.856900    5.262250    2.944000   
50%     14.355000   14.320000      0.873450    5.523500    3.237000   
75%     17.305000   15.715000      0.887775    5.979750    3.561750   
max     21.180000   17.250000      0.918300    6.675000    4.033000   

       coefficient   length_groove   type_wheat  
count    210.000000      210.000000  210.000000  
mean       3.700201        5.408071    2.000000  
std        1.503557        0.491480    0.818448  
min        0.765100        4.519000    1.000000  
25%        2.561500        5.045000    1.000000  
50%        3.599000        5.223000    2.000000  
75%        4.768750        5.877000    3.000000  
max        8.456000        6.550000    3.000000

In [52]:
from sklearn.naive_bayes import GaussianNB 

In [53]:
# Instanciamos nuestro modelo 
modelo = GaussianNB()

In [58]:
# Retiramos la variable que vamos a predecir
X = dataset.drop(['type_wheat'], axis=1) 
y = dataset['type_wheat']
dataset.head(3)

area      perimeter   compactness   length   width   coefficient   \
0     15.26       14.84        0.8710    5.763   3.312         2.221   
1     14.88       14.57        0.8811    5.554   3.333         1.018   
2     14.29       14.09        0.9050    5.291   3.337         2.699   

   length_groove   type_wheat  
0           5.220           1  
1           4.956           1  
2           4.825           1

In [59]:
# Separamos los datasets de entrenamiento y de testing
from sklearn.model_selection import train_test_split
Xtrain, Xtest, ytrain, ytest = train_test_split(X, y,random_state=2)

In [64]:
#Una vez dividido entrenamos nuestro Modelo.
modelo.fit(Xtrain, ytrain)

GaussianNB()

In [65]:
#Procedemos a hacer la prediccion de las etiquetas.
ypred = modelo.predict(Xtest)
ypred

array([1, 3, 1, 3, 1, 3, 1, 1, 1, 3, 1, 3, 3, 3, 2, 2, 1, 3, 3, 2, 1, 2,
       1, 3, 1, 1, 2, 1, 1, 1, 1, 2, 1, 2, 1, 2, 1, 2, 1, 1, 1, 3, 1, 2,
       3, 1, 3, 2, 2, 3, 2, 2, 1])

In [67]:
# Evaluamos la performance de nuestro modelo.
from sklearn.metrics import accuracy_score 
round(accuracy_score(ytest, ypred), 2)

0.87

Obtenemos un valor de accuracy del 87%, es decir que 87 de cada 100 casos fueron correctamente predichos.
